pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
+                devices, deprecated_devices_specific_flag, num_processes, gpus, ipus, tpu_cores
+            )
 
-    @property
-    def use_ipu(self) -> bool:
-        return self._accelerator_type == _AcceleratorType.IPU and self.has_ipu
+        if self._device_flag == "auto" and self._accelerator_flag is None:
+            raise MisconfigurationException(
+                f"You passed `devices={devices}` but haven't specified"
+                " `accelerator=('auto'|'tpu'|'gpu'|'ipu'|'cpu')` for the devices mapping"
+            )
 
-    def _set_devices_to_cpu_num_processes(self) -> None:
-        if self.num_processes == 1:
-            self._map_devices_to_accelerator(_AcceleratorType.CPU)
+    def _map_deprecated_devices_specfic_info_to_accelerator_and_device_flag(
+        self,
+        devices: Optional[Union[List[int], str, int]],
+        deprecated_devices_specific_flag: Union[int, List[int]],
+        num_processes: Optional[int],
+        gpus: Optional[List[int]],
+        ipus: Optional[int],
+        tpu_cores: Optional[Union[int, List[int]]],
+    ) -> None:
+        """Sets the `device_flag` based on num_processes, gpus, ipus, tpu_cores."""
+        if devices:
+            # TODO: @awaelchli improve error message
+            rank_zero_warn(
+                f"The flag `devices={devices}` will be ignored, "
+                f"instead the device specific number {deprecated_devices_specific_flag} will be used"
+            )
 
-    def _map_devices_to_accelerator(self, accelerator: str) -> bool:
-        if self.devices is None:
-            return False
-        if accelerator == _AcceleratorType.TPU and _TPU_AVAILABLE:
-            if self.devices == "auto":
-                self.devices = TPUAccelerator.auto_device_count()
-            self.tpu_cores = device_parser.parse_tpu_cores(self.devices)
-            return True
-        if accelerator == _AcceleratorType.IPU and _IPU_AVAILABLE:
-            if self.devices == "auto":
-                self.devices = IPUAccelerator.auto_device_count()
-            self.ipus = self.devices
-            return True
-        if accelerator == _AcceleratorType.GPU and torch.cuda.is_available():
-            if self.devices == "auto":
-                self.devices = GPUAccelerator.auto_device_count()
-            self.gpus = self.devices
-            self.parallel_device_ids = device_parser.parse_gpu_ids(self.devices)
-            return True
-        if accelerator == _AcceleratorType.CPU:
-            if self.devices == "auto":
-                self.devices = CPUAccelerator.auto_device_count()
-            if not isinstance(self.devices, int):
-                raise MisconfigurationException(
+        if [(num_processes is not None), (gpus is not None), (ipus is not None), (tpu_cores is not None)].count(
+            True
+        ) > 1:
+            # TODO: @awaelchli improve error message
+            rank_zero_warn("more than one device specific flag has been set")
```

Comment:
These all will be deprecated soon, I think better keep the existing behavior? I think this is an error because if user set both num_process and gpu, we use num_processes and cpuAccelerator.